src/types/package.go

```diff
@@ -20,11 +22,24 @@ type ZarfMetadata struct {
 	Architecture string `yaml:"architecture,omitempty" jsonschema:"description=The target cluster architecture of this package"`
 }
 
-// ZarfBuildData is written during the packager.Create() operation to track details of the created package
+// ZarfBuildData is written during the packager.Create() operation to track details of the created package.
 type ZarfBuildData struct {
 	Terminal     string `yaml:"terminal"`
 	User         string `yaml:"user"`
 	Architecture string `yaml:"architecture"`
 	Timestamp    string `yaml:"timestamp"`
 	Version      string `yaml:"version"`
 }
+
+// ZarfPackageVariable are variables that can be used to dynaically template K8s resources.
+type ZarfPackageVariable struct {
+	Name    string  `yaml:"name" jsonschema:"description=The name to be used for the variable,pattern=^[A-Z_]+$"`
+	Default *string `yaml:"default,omitempty" jsonschema:"description=The default value to use for the variable"`
```

Comment:
Can you explain why we need a pointer here?  It seems that the null value of "" would be fine to work with vs checking the pointer for nil.

Reply:
That depends on if we want to not template non-defaulted non-prompted values or not.  If a user does not specify something for one and default doesn't have the ability to be nil then you couldn't tell the difference between an unspecified `""` or an actual value defaulted to `""`.  This is probably an edge case though and we could just say everything is defaulted to "".

Reply:
Removed the pointer here.